middleware/middleware.go

```diff
@@ -2,44 +2,68 @@ package middleware
 
 import (
 	"database/sql"
+	"fmt"
 	"log"
 
-	"github.com/Necroforger/dgrouter/exrouter"
+	"github.com/lus/dgc"
 	"github.com/wpi-25/scribe/db"
 )
 
-func AdminOnly(fn exrouter.HandlerFunc) exrouter.HandlerFunc {
-	return func(ctx *exrouter.Context) {
-		log.Println("Attempting to authenticate an admin command")
-
-		row := db.DB.QueryRow("SELECT min_role_id FROM guild_settings WHERE guild_id = ?", ctx.Msg.GuildID)
-		var min_role_id sql.NullString
-		err := row.Scan(min_role_id)
-		if err != nil {
-			ctx.Reply("Could not Scan: %s", err)
+func AdminOnly(next dgc.ExecutionHandler) dgc.ExecutionHandler {
+	return func(c *dgc.Ctx) {
+		is_admin := false
+		for _, flag := range c.Command.Flags {
+			if flag == "admin" {
+				is_admin = true
+			}
 		}
-		err = row.Err()
-		if err != nil {
-			guild, _ := ctx.Guild(ctx.Msg.GuildID)
-			owner := guild.OwnerID
-			if owner == ctx.Msg.Author.ID {
-				fn(ctx)
-			} else {
-				ctx.Reply("Could not authenticate you! Have the server owner set an admin role.")
+		if is_admin {
+			log.Println("Attempting to authenticate an admin command")
+
+			row := db.DB.QueryRow("SELECT min_role_id FROM guild_settings WHERE guild_id = ?", c.Event.GuildID)
+			var min_role_id sql.NullString
+			err := row.Scan(min_role_id)
+			if err != nil {
+				c.RespondText(fmt.Sprintf("Could not Scan: %s", err))
+				return
+			}
+			err = row.Err()
+			if err != nil {
+				guild, _ := c.Session.Guild(c.Event.GuildID)
+				owner := guild.OwnerID
+				if owner == c.Event.Author.ID {
```

Comment:
This will only allow the owner of the guild to use an admin command? Do we want this to be for anyone with the "Administrator" privilege?

Reply:
Seeing the next function, yes it does look like this is the functionality we want.

Reply:
I thought this was going to be anyone with the role as defined by `minperms`?

Reply:
In my original idea, `minperms` was going to determine who can and cannot create certain types of invites. Admin commands were gonna be locked to people with "Create Invites" (I wasn't looking at the doc earlier)

Reply:
Ohhhh, I misunderstood what you meant in the doc. To be honest, I think that we should just restrict the creation of infinite invites to admins with `minperms`

Reply:
Works for me